erpnext/stock/doctype/packing_slip/packing_slip.py

```diff
@@ -920,6 +924,19 @@ def get_unpack_transfer_sles(self, sl_entries):
 
 			sl_entries.append(sle_in)
 
+	def get_rejected_items_sles(self, sl_entries):
+		for d in self.get("items"):
+			sle_out = self.get_sl_entries(d, {
+				"warehouse": d.source_warehouse,
+				"actual_qty": -flt(d.rejected_qty),
+			})
+			sl_entries.append(sle_out)
+			sle_in = self.get_sl_entries(d, {
+				"warehouse": frappe.db.get_single_value('Stock Settings', 'default_rejected_warehouse'),
```

Comment:
unsafe to use settings value because the value can change in the future and then break the system

- Rejected Warehouse field should be added in Packing Slip parent DocType
- Rejected Warehouse should be auto filled from settings in client side and also editable by user
- Rejected Warehouse should be validated: not empty if rejected qty is set